src/Symfony/Component/Form/Tests/ButtonBuilderTest.php

```diff
@@ -40,6 +39,14 @@ public function testValidNames($name)
         $this->assertInstanceOf('\Symfony\Component\Form\ButtonBuilder', new ButtonBuilder($name));
     }
 
+    /**
+     * @group legacy
+     */
+    public function testNameContainingSquareBrackets()
```

Comment:
`testNameContainingIllegalCharacters` would be better here IMHO as it explains _why_ those brackets are there.